frontend/components/bc/BcInputSearch.vue

```diff
@@ -0,0 +1,322 @@
+<script setup lang="ts" generic="T">
+const props = withDefaults(defineProps<{
+  hasError?: boolean,
+  hasFocus?: boolean,
+  isLoading: boolean,
+  results: T[] | undefined,
+  shouldClearOnSubmit?: boolean,
+  shouldSelectFirstResult?: boolean,
+}>(), {
+  shouldClearOnSubmit: true,
+})
+
+const emit = defineEmits<{
+  (e: 'search', value: string): void,
+  (e: 'submit', value: T): void,
+}>()
+
+const { t: $t } = useTranslation()
+
+const input = defineModel<string>()
+
+const elementInput = ref<HTMLElement >()
+const hasInput = computed(() => input.value?.length)
+
+const idResults = useId()
+const elementResults = ref<HTMLElement>()
+
+const idElementList = useId()
+const elementList = ref<Array<HTMLElement>>()
+
+const resultItems = computed(() => {
```

Comment:
Isn't this the same as `results.length`?

Reply:
true...thx 💯